osu.Game/OsuGame.cs

```diff
@@ -295,6 +295,8 @@ private void beatmapChanged(ValueChangedEvent<WorkingBeatmap> beatmap)
             var nextBeatmap = beatmap.NewValue;
             if (nextBeatmap?.Track != null)
                 nextBeatmap.Track.Completed += currentTrackCompleted;
+
+            nextBeatmap?.LoadBeatmapAsync();
```

Comment:
Could probably invoke the disposal on the previous beatmap before it's changed? Otherwise we rely only on the finalizer to run for the task to get cancelled.